template/gitlab-runner.tpl

```diff
@@ -176,6 +170,12 @@ chkconfig --add remove_gitlab_registration
 # As noted above, this does nothing more than make the lockfile.
 service remove_gitlab_registration start
 
+if ! ( rpm -q gitlab-runner >/dev/null )
+then
+  curl --fail --retry 6 -L https://packages.gitlab.com/install/repositories/runner/gitlab-runner/script.rpm.sh | bash
+  yum install gitlab-runner-${gitlab_runner_version} -y
+fi
+
 ${post_install}
 
 service gitlab-runner restart
```

Comment:
suggestion: change this to (a redundant) `gitlab-runner start`. A restart might cause the runner to lose the jobs it has picked up already.

```suggestion
service gitlab-runner start
```

Reply:
Let's go one step further: If it's redundant, we can remove it as it doesn't do anything. If it is there due to the `${post_install}`: As this script is under user control it should also start/restart, ... the Runner if needed.

Just tested this PR and checked the logs. After the `yum` command and before any other command in the startup script, I read: `Feb 23 07:59:31 ip-100-64-24-219 systemd: Started GitLab Runner.`

Reply:
I'm just a bit afraid of [Chesterton's fence](https://en.wiktionary.org/wiki/Chesterton%27s_fence) 😅. Especially since I just started using this module.

Reply:
Just tested it this way, notice the sleep as stand in for long post_install command.

```shell
if ! ( rpm -q gitlab-runner >/dev/null )
then
  curl --fail --retry 6 -L https://packages.gitlab.com/install/repositories/runner/gitlab-runner/script.rpm.sh | bash
  yum install gitlab-runner-${gitlab_runner_version} -y
fi

${post_install}

sleep 60

service gitlab-runner restart
```

Resulted in job error, not loss. But still, seems very bad.

![image](https://user-images.githubusercontent.com/17970041/220853246-0e71c097-97c7-4c65-a51b-4fac82ae1699.png)

Reply:
Also true. But as far as I can see, the commit is from 2017, first installing the GitLab Runner, then the docker+machine and then restarting the GitLab Runner.

As we know from the new logs, the GitLab Runner is up and running. Jobs are processed. So I am fine with this change. @npalm Do you remember why this restart is there?

Reply:
Checked the docs for [manual installation](https://docs.gitlab.com/runner/install/linux-manually.html#using-binary-file) of the GitLab Runner. They install the Runner and starts it. That's all. No second start/restart at all.